android-components/components/support/base/build.gradle

```diff
@@ -5,7 +5,7 @@
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
-def generatedSrcDir = new File(buildDir, "generated/components/src/main/java")
+def generatedSrcDir = new File(layout.buildDirectory.get().asFile, "generated/components/src/main/java")
```

Comment:
Just out of curiosity: Wouldn't `layout.buildDirectory` work just as well here?

Reply:
That would throw a runtime exception as` buildDirectory` is a `DirectoryProperty` and there is no `File` constructor that takes that combination of arguments.